internal/controller/scheduler/scheduler.go

```diff
@@ -718,6 +654,79 @@ func PatchPodSpec(original *corev1.PodSpec, patch *corev1.PodSpec) (*corev1.PodS
 	return &patchedSpec, nil
 }
 
+func (w *worker) createInitContainer(podSpec *corev1.PodSpec, workspaceVolume *corev1.Volume) corev1.Container {
```

Comment:
Since init containers are a broader topic than the single init container returned here:

```suggestion
func (w *worker) createWorkspaceSetupContainer(podSpec *corev1.PodSpec, workspaceVolume *corev1.Volume) corev1.Container {
```

or maybe `createCopyAgentContainer` ?